recipes/gsl/all/conanfile.py

```diff
@@ -0,0 +1,158 @@
+        return "source_subfolder"
+
+    @property
+    def _settings_build(self):
+        return getattr(self, "settings_build", self.settings)
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            del self.options.fPIC
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True,
+                  destination=self._source_subfolder)
+
+    def build_requirements(self):
+        self.build_requires("libtool/2.4.6")
+        if tools.os_info.is_windows and not tools.get_env("CONAN_BASH_PATH"):
+            self.build_requires("msys2/cci.latest")
+
+    @contextmanager
+    def _build_context(self):
+        if self.settings.compiler == "Visual Studio":
+            with tools.vcvars(self):
+                env = {
+                    "CC": "cl -nologo",
+                    "CXX": "cl -nologo",
+                    "LD": "link -nologo",
+                    "AR": "{} lib".format(tools.unix_path(self.deps_user_info["automake"].ar_lib)),
+                }
+                with tools.environment_append(env):
+                    yield
+        else:
+            yield
+
+    def _patch_source(self):
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
+        with tools.chdir(self._source_subfolder):
+            self.run("{} -fiv".format(tools.get_env("AUTORECONF")), win_bash=tools.os_info.is_windows, run_environment=True)
+        tools.replace_in_file(os.path.join(self._source_subfolder, "configure"),
+                              r"-install_name \$rpath/",
+                              "-install_name @rpath/")
+
+    def _configure_autotools(self):
+        if self._autotools:
+            return self._autotools
+        self._autotools = AutoToolsBuildEnvironment(self, win_bash=tools.os_info.is_windows)
+
+        self._autotools.libs = []
+        yes_no = lambda v: "yes" if v else "no"
+        args = [
+            "--enable-shared={}".format(yes_no(self.options.shared)),
+            "--enable-static={}".format(yes_no(not self.options.shared)),
+        ]
+        if self.settings.os == "Windows":
+            self._autotools.defines.extend(["HAVE_WIN_IEEE_INTERFACE", "WIN32"])
+            if self.options.shared:
+                self._autotools.defines.append("GSL_DLL")
+        elif self.settings.os == "Linux":
+            self._autotools.defines.append("HAVE_GNUX86_IEEE_INTERFACE")
+
+        if self.settings.compiler == "Visual Studio":
+            self._autotools.flags.append("-FS")
+            self._autotools.cxx_flags.append("-EHsc")
+        args.extend([
+            "ac_cv_func_memcpy=yes",
+            "ac_cv_func_memmove=yes",
+            "ac_cv_c_c99inline=no",
+        ])
```

Comment:
These were only needed for MSVC.
Looks like they don't harm other compiler though.
```suggestion
            args.extend([
                "ac_cv_func_memcpy=yes",
                "ac_cv_func_memmove=yes",
                "ac_cv_c_c99inline=no",
            ])
```